src/enhanced-switch.jsx

```diff
@@ -400,55 +359,72 @@ const EnhancedSwitch = React.createClass({
     const focusRipple = (
       <FocusRipple
         key="focusRipple"
-        innerStyle={rippleStyle}
-        color={rippleStyle.color}
+        innerStyle={mergedRippleStyle}
+        color={mergedRippleStyle.color}
         muiTheme={this.state.muiTheme}
         show={this.state.isKeyboardFocused}
       />
     );
 
     const ripples = [
-      hideTouchRipple ? null : touchRipple,
-      this.props.disabled || disableFocusRipple ? null : focusRipple,
+      showTouchRipple ? touchRipple : null,
+      showFocusRipple ? focusRipple : null,
     ];
 
+    const inputElement = (
+      <input
+        {...other}
+        ref="checkbox"
+        id={inputId}
+        type={inputType}
+        style={prepareStyles(Object.assign(styles.input, inputStyle))}
+        name={name}
+        value={value}
+        defaultChecked={defaultSwitched}
+        disabled={disabled}
+        onBlur={this._handleBlur}
+        onFocus={this._handleFocus}
+        onChange={this._handleChange}
+        onMouseUp={showTouchRipple && this._handleMouseUp}
+        onMouseDown={showTouchRipple && this._handleMouseDown}
+        onMouseLeave={showTouchRipple && this._handleMouseLeave}
+        onTouchStart={showTouchRipple && this._handleTouchStart}
+        onTouchEnd={showTouchRipple && this._handleTouchEnd}
+      />
+    );
+
     // If toggle component (indicated by whether the style includes thumb) manually lay out
     // elements in order to nest ripple elements
-    const switchElement = !this.props.thumbStyle ? (
+    const switchOrThumbElement = !thumbStyle ? (
       <div style={prepareStyles(wrapStyles)}>
-        {this.props.switchElement}
+        {switchElement}
         {ripples}
       </div>
     ) : (
       <div style={prepareStyles(wrapStyles)}>
-        <div style={prepareStyles(Object.assign({}, this.props.trackStyle))} />
-        <Paper style={this.props.thumbStyle} zDepth={1} circle={true}> {ripples} </Paper>
+        <div style={prepareStyles(Object.assign({}, trackStyle))} />
+        <Paper style={thumbStyle} zDepth={1} circle={true}> {ripples} </Paper>
       </div>
     );
 
-    const labelPositionExist = this.props.labelPosition;
-
-    // Position is left if not defined or invalid.
-    const elementsInOrder = (labelPositionExist &&
-      (this.props.labelPosition.toUpperCase() === 'RIGHT')) ? (
-      <ClearFix style={styles.controls}>
-        {switchElement}
+    const elementsInOrder = labelPosition === 'right' ? (
```

Comment:
the propType validation will throw warning if it's not either `left` or `right`, insensitive comparison in unnecessary.